fe/fe-core/src/main/java/com/starrocks/persist/metablock/SRMetaBlockReader.java

```diff
@@ -22,13 +22,32 @@ public interface SRMetaBlockReader {
 
     SRMetaBlockHeader getHeader();
 
-    <T> T readJson(Class<T> returnClass) throws IOException, SRMetaBlockEOFException;
-
     int readInt() throws IOException, SRMetaBlockEOFException;
 
     long readLong() throws IOException, SRMetaBlockEOFException;
 
-    Object readJson(Type returnType) throws IOException, SRMetaBlockEOFException;
+    byte readByte() throws IOException, SRMetaBlockEOFException;
+
+    short readShort() throws IOException, SRMetaBlockEOFException;
+
+    double readDouble() throws IOException, SRMetaBlockEOFException;
+
+    float readFloat() throws IOException, SRMetaBlockEOFException;
+
+    char readChar() throws IOException, SRMetaBlockEOFException;
+
+    boolean readBoolean() throws IOException, SRMetaBlockEOFException;
+
+    String readString() throws IOException, SRMetaBlockEOFException;
+
+    <T> T readJson(Type returnType) throws IOException, SRMetaBlockEOFException;
+
+    <T> T readJson(Class<T> classType) throws IOException, SRMetaBlockEOFException;
+
+    <T> void readCollection(Class<T> classType, CollectionConsumer<? super T> action) throws IOException, SRMetaBlockEOFException;
+
+    <K, V> void readMapEntry(Type keyType, Type valueType, MapEntryConsumer<? super K, ? super V> action)
```

Comment:
readMap？

Reply:
add `writeCollection` and `writeMap` in writer?
The logic needs to be consistent with the reader, write size then write elements, read size then elements.

Reply:
Yes, writeCollection and writeMap will make the api consistent, but in some cases it is not easy to get a simple collection, and the process of write is complicated. For example:  LocalMetastore will write a database object and a list of its tables for every database.

We can just give the basic api of writer, and this will give developers more freedom.